docs/csharp/local-functions-vs-lambdas.md

```diff
@@ -15,44 +15,50 @@ ms.assetid: 368d1752-3659-489a-97b4-f15d87e49ae3
 
 ### Local functions compared to Lambda expressions
 
-In some use cases, you could create a lambda expression and use it
-without needing to create a local function. Here's an example async
-method that does just that:
+At first glance, local functions and lambda expressions are very similar.
+Depending on your needs, local functions may be a much better and simpler
+solution.
 
-[!code-csharp[TaskLambdaExample](../../samples/snippets/csharp/new-in-7/AsyncWork.cs#36_TaskLambdaExample "Task returning method with lambda expression")]
+Let's examine the differences between the local function and lambda expression
+implementations of the factorial algorithm. First the version using a local function:
 
-However, there are a number of reasons to prefer using local functions
-instead of defining and calling lambda expressions.
+[!code-csharp[LocalFunctionFactorial](../../samples/snippets/csharp/new-in-7/MathUtilities.cs#37_LocalFunctionFactorial "Recursive factorial using local function")]
 
-First, for lambda expressions, the compiler must create an anonymous class
-and an instance of that class to store any variables captured by the
-closure. The closure for this lambda expression contains the `address`,
-`index` and `name` variables. 
+Contrast that implementation with a version that uses lambda expressions:
 
-Second, lambda expressions are implemented by instantiating a delegate
+[!code-csharp[26_LambdaFactorial](../../samples/snippets/csharp/new-in-7/MathUtilities.cs#38_LambdaFactorial "Recursive factorial using lambda expressions")]
+
+First, lambda expressions are implemented by instantiating a delegate
 and invoking that delegate. Local functions are implemented as method calls.
 The instantiation necessary for lambda expressions means extra memory
 allocations, which may be a performance factor in time critical code paths.
-Local functions do not incur this overhead.
+Local functions do not incur this overhead. Because of the recursive nature
+of this factorial algorithm, the number of allocations saved is on the order
+of the value of the argument to the public method.
+
+Second, for lambda expressions, the compiler must create an anonymous class
+and an instance of that class to store any variables captured by the
+closure. The closure for this lambda expression contains the `address`,
+`index` and `name` variables. In the case of local functions, the object
+that implements the closure may be a `struct` type. That would save on
+an allocation.
```

Comment:
In this specific case, the local function does not close over any variables, so it does not have a closure object at all (`struct` or otherwise). Though the general statement is correct, so I'm not sure if this needs to be changed.